mirascope/tools/_filesystem.py

```diff
@@ -0,0 +1,215 @@
+from pathlib import Path
+from typing import ClassVar
+
+from pydantic import Field, field_validator
+
+from mirascope.core.base import toolkit_tool
+from mirascope.tools.base import ConfigurableToolKit, _ToolConfig
+
+
+class FileSystemToolkitConfig(_ToolConfig):
+    """Configuration for filesystem toolkit"""
+
+    max_file_size: int = Field(
+        default=10 * 1024 * 1024,  # 10MB
+        description="Maximum file size in bytes",
+    )
+    allowed_extensions: list[str] = Field(
+        default=["txt", "md", "csv", "json", "yml", "yaml"],
+        description="List of allowed file extensions",
+    )
+
+
+class FileSystemToolkit(ConfigurableToolKit[FileSystemToolkitConfig]):
+    """Toolkit for filesystem operations."""
+
+    __config__ = FileSystemToolkitConfig()
+    __namespace__ = "filesystem"
+    __prompt_usage_description__: ClassVar[str] = """
+    - Tools for filesystem operations:
+        - read_file: Reads content from a file
+        - write_file: Writes content to a file
+        - list_directory: Lists directory contents
+        - create_directory: Creates a new directory
+        - delete_file: Deletes a file
+    """
+
+    base_directory: Path = Field(
+        default=Path.cwd(), description="Base directory for file operations"
+    )
+
+    @field_validator("base_directory")
+    def validate_base_directory(cls, v: Path) -> Path:
+        if not v.exists():
+            raise ValueError(f"Base directory {v} does not exist")
+        if not v.is_dir():
+            raise ValueError(f"{v} is not a directory")
+        return v
+
+    def _validate_path(self, path: str) -> str | None:
+        """Validates file path and returns error message if invalid."""
+        file_path = self.base_directory / path
+
+        # Check for path traversal attempts
+        try:
+            file_path.resolve().relative_to(self.base_directory.resolve())
+        except ValueError:
+            return "Error: Invalid path (attempted path traversal)"
+
+        return None
+
+    def _validate_extension(self, path: str) -> str | None:
+        """Validates file extension and returns error message if invalid."""
+        extension = Path(path).suffix.lstrip(".")
+        if extension not in self._config().allowed_extensions:
+            return f"Error: Invalid file extension. Allowed: {self._config().allowed_extensions}"
+        return None
+
+    @toolkit_tool
+    def read_file(self, path: str) -> str:
```

Comment:
I feel like best practice would be to implement this as a class instead (i.e. `ReadFile`) so that the user can potentially do something like `if isinstance(tool, FileSystemToolkit.ReadFile): ...` right?